controllers/numaresourcesoperator_controller.go

```diff
@@ -769,7 +769,7 @@ func daemonsetUpdater(mcpName string, gdm *rtestate.GeneratedDesiredManifest) er
 }
 
 func isDaemonSetReady(ds *appsv1.DaemonSet) bool {
-	ok := (ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady)
+	ok := (ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady) || ds.Status.DesiredNumberScheduled == 0
```

Comment:
please split this complex logic in some simpler `if` blocks, adding subfunctions if it helps